### ISSUE TITLE
Shared storage support

### ISSUE THREAD

**Issue by nielsAD** [jybp/casc#1]
For future versions, Warcraft 3 will start to [merge](https://us.forums.blizzard.com/en/warcraft3/t/adding-shared-container-tech-with-version-1326/24419) the storage directories for different releases of the game (e.g. Retail and PTR). I assume this will mean that people who have installed multiple versions will have multiple entries in their `.build.info` file (?). Currently, the library [assumes](https://github.com/jybp/casc/blob/master/local.go#L43) that there will be one entry at most.

Maybe `casc.Local()` can be altered to accept a product code (w3 vs w3t), or maybe the library can be altered to read `.flavor.info` files, so that users can pass `C:\Program Files\Warcraft III\_retail_`  and the product code / data path will be inferred (preserving backwards compatibility for the library API).

More info: https://wowdev.wiki/TACT#Shared_storage

**Comment by jybp** [comment]
Thank you for bringing this issue to my attention @nielsAD 
I will definitely look into it and add support to the new layout.

**Comment by jybp** [comment]
@nielsAD Do you know how to enable this "shared" storage feature so I can reproduce the issue?

I just downloaded retail war3 using battle.net on MacOS. I do have the `_retail_` folder but there still is a root `Data` folder:

```
$ ls -1a
.
..
.build.info
.patch.result
.product.db
Data
Launcher.db
Warcraft III Launcher.app
_retail_
```

casc v1.0 still seem to extract files properly:

```
$ echo "War3.w3mod:Movies/HumanEd_1280.avi" | go run cmd/casc/main.go -dir "/Applications/Warcraft III"
2020/06/26 09:00:51 HumanEd_1280.avi
```

When trying to install the PTR, it defaults to another folder`/Applications/Warcraft III Public Test/`.
Choosing to install the PTR at location `/Applications/Warcraft III` simply creates a whole new `Warcraft III Public Test` subfolder.

```
$ ls -1a
.
..
.DS_Store
.build.info
.patch.result
.product.db
Data
Launcher.db
Warcraft III Launcher.app
Warcraft III Public Test
_retail_
```

Something is off. Looks like I'm in the same spot as some of the people in your linked forum post.

**Comment by nielsAD** [comment]
Yeah, currently everything is still fine, I just figured I would open this issue ahead of time. The forum post ends with 

> Once the 1.32.6 Retail version is released, we highly recommend that you uninstall the 1.32.6 PTR version. Any future PTR or Retail releases after 1.32.6 will now be able to share the data directory if desired.

so shared storage will be available starting with the next PTR release. Not sure when that will be, I assume somewhere in the next month or so.

**Comment by jybp** [comment]
Oh I see. I guess I'll just wait until it's in actual use if that's fine with you.
Do not hesitate to ping me in this issue once it's in use.

I see you're using this pkg in https://github.com/nielsAD/gowarcraft3 Very nice :)

**Comment by nielsAD** [comment]
Yeah, it's been working great so far!

Coincidentally, there has been a PTR patch last Monday, and reinstalling PTR did indeed enable shared storage for me. My directory layout is currently:

```
├── .build.info
├── Data
│   ├── config
│   ├── data
│   └── indices
├── Launcher.db
├── .patch.result
├── .product.db
├── _ptr_
│   ├── .flavor.info
│   └── x86_64
├── _retail_
│   ├── .flavor.info
│   └── x86_64
└── Warcraft III Launcher.exe
```

with `.build.info` listing both versions. The package does not give any errors, but it just picks the last mentioned version from the build file at the moment (PTR in my case).

`.build.info`:
```
Branch!STRING:0|Active!DEC:1|Build Key!HEX:16|CDN Key!HEX:16|Install Key!HEX:16|IM Size!DEC:4|CDN Path!STRING:0|CDN Hosts!STRING:0|CDN Servers!STRING:0|Tags!STRING:0|Armadillo!STRING:0|Last Activated!STRING:0|Version!STRING:0|Product!STRING:0
eu|1|733e8f4a3e8e0feaa44d52b458592651|d0427daa9162695282f0daeffb46b1d1|6de823310e532477e16e1fc5832506e3||tpr/war3|level3.blizzard.com eu.cdn.blizzard.com|http://eu.cdn.blizzard.com/?maxhosts=4 http://level3.blizzard.com/?maxhosts=4 https://blzddist1-a.akamaihd.net/?fallback=1&maxhosts=4 https://eu.cdn.blizzard.com/?fallback=1&maxhosts=4 https://level3.ssl.blizzard.com/?fallback=1&maxhosts=4|Windows EU? acct-NLD? geoip-NL? enUS speech?:Windows EU? acct-NLD? geoip-NL? enUS text?|||1.32.6.15355|w3
eu|1|b5789e1d3f34ffb8a19b9273166d55c0|d0427daa9162695282f0daeffb46b1d1|a30f94ab15aabd49492f0925e41b8e12||tpr/war3|eu.cdn.blizzard.com level3.blizzard.com|http://eu.cdn.blizzard.com/?maxhosts=4 http://level3.blizzard.com/?maxhosts=4 https://blzddist1-a.akamaihd.net/?fallback=1&maxhosts=4 https://eu.cdn.blizzard.com/?fallback=1&maxhosts=4 https://level3.ssl.blizzard.com/?fallback=1&maxhosts=4|Windows EU? acct-NLD? geoip-NL? enUS speech?:Windows EU? acct-NLD? geoip-NL? enUS text?|||1.32.7.15539|w3t
```

**Comment by jybp** [comment]
Yes it just picks the last line arbitrarily and might select the wrong version.
I just pushed an update `v1.2` that selects the correct version if the "Shared Storage" feature is enabled. (Only 'retails' are supported atm, so 'w3' is selected in Warcraft III's case).

I did not add the possibility to parse another version like the  "PTR" version. That's a feature that could be added if you really need it.
